src/metpy/calc/thermo.py

```diff
@@ -3264,3 +3264,66 @@ def gradient_richardson_number(height, potential_temperature, u, v, vertical_dim
     dvdz = first_derivative(v, x=height, axis=vertical_dim)
 
     return (mpconsts.g / potential_temperature) * (dthetadz / (dudz ** 2 + dvdz ** 2))
+
+@exporter.export
+@preprocess_and_wrap()
+@check_units('[pressure]', '[temperature]', '[temperature]')
+def showalter_index(pressure, temperature, dewpt): 
+    """Calculate Showalter Index from pressure temperature and 850 hPa lcl
+    
+    Showalter Index derived from [Galway1956]_:
+    SI = T500 - Tp500
+    
+    where:
+    T500 is the measured temperature at 500 hPa
+    Tp500 is the temperature of the lifted parcel at 500 hPa
+    
+   Parameters
+   ----------
+        
+        pressure : `pint.Quantity`
+            Atmospheric pressure level(s) of interest, in order from highest to
+        lowest pressure
+        
+        temperature : `pint.Quantity`
+            Parcel temperature for corresponding pressure 
+        
+        dewpt (:class: `pint.Quantity`):
+            Parcel dew point temperatures for corresponding pressure
+        
+
+    Returns
+    -------
+    `pint.Quantity`
+        Showalter index in delta degrees celsius
+        
+    """
+
+    # find the measured temperature and dew point temperature at 850 hPa.
+    idx850 = np.where(pressure == 850 * units.hPa)
+    T850 = temperature[idx850]
+    Td850 = dewpt[idx850]
```

Comment:
I think this will work a little better for more data sources if we use `interpolate_1d` from `metpy.calc.tools`:
```suggestion
    T850, Td850 = interpolate_1d(850 * units.hPa, pressure, temperature, dewpt)
```